activerecord/test/cases/associations/belongs_to_associations_test.rb

```diff
@@ -1205,6 +1205,14 @@ def test_reassigning_the_parent_id_updates_the_object
     assert_equal companies(:another_firm), client.firm_with_condition
   end
 
+  def test_assigning_nil_on_an_association_clears_the_associations_inverse
+    with_has_many_inversing do
+      book     = Book.create!
+      citation = book.citations.create!
+      citation.book = nil
```

Comment:
Is this missing an assertion?

Reply:
Not really, this hard crash on `main`. But yes, I could add some assertions.

Reply:
I think an assertion that the inverse is cleared would be nice, but this PR looks good to me.

Reply:
Actually that's not something that was ever supported as far as I can tell, and it seems very hard to support.

I think the fix is just to not do anything on `has_many`.

Reply:
That makes sense...